cli/flox/src/commands/show.rs

```diff
@@ -126,7 +137,70 @@ fn construct_show_params(
     Ok(search_params)
 }
 
-fn render_show(search_results: &[SearchResult]) -> Result<()> {
+fn render_show_catalog(
+    search_results: &[SearchResult],
+    expected_systems: &HashSet<System>,
+) -> Result<()> {
+    if search_results.is_empty() {
+        // This should never happen since we've already checked that the
+        // set of results is non-empty.
+        bail!("no packages found");
+    }
+    let pkg_name = search_results[0].rel_path.join(".");
+    let description = search_results[0]
+        .description
+        .as_ref()
+        .map(|d| d.replace('\n', " "))
+        .unwrap_or(DEFAULT_DESCRIPTION.into());
+    println!("{pkg_name} - {description}");
+
+    // Organize the versions to be queried and printed
+    let version_to_systems = {
+        let mut map = BTreeMap::new();
+        for pkg in search_results.iter() {
+            if let Some(ref version) = pkg.version {
+                map.entry(version.clone())
+                    .or_insert(HashSet::new())
+                    .insert(pkg.system.clone());
+                // map.insert(version.clone(), pkg.system.clone());
+            }
+        }
+        map
+    };
+    let mut seen_versions = HashSet::new();
+    for pkg in search_results {
+        if let Some(ref version) = pkg.version {
```

Comment:
nonblocking: I think all the info you need is in `version_to_systems`, how come you aren't iterating over that?

Reply:
The sorting in the BTreeMap may not be the order in which results were returned to us so I preserve that order rather than purely lexical sorting that you'd get with the BTreeMap.

Reply:
It makes sense to me that the server controls ordering for search, but I'm not sure about it for show 🤔 Seems like we probably want to sort by version no matter what, and that feels a bit more like a client than server responsibility. But I guess maybe the server should control all version parsing/sorting